src/sentry/web/api.py

```diff
@@ -175,6 +183,9 @@ def _dispatch(self, request, helper, project_id=None, origin=None, *args, **kwar
             if not project:
                 raise APIError('Client must be upgraded for CORS support')
             if not is_valid_origin(origin, project):
+                if project:
+                    tsdb.incr(tsdb.models.project_total_received_cors,
```

Comment:
Kind of odd that this happens in two places

Reply:
Do you mean that the `if project` check is extraneous or are you referring to the `is_valid_origin` call?

Reply:
Both, really. It's not an issue with this change, I just think it's a little odd in general but didn't really do enough research to know if there's a better approach (which again, would be totally unrelated to this change… I'm just musing.)